File: load_moread_apter/src/main/java/me/simple/loadmoreadapter/LoadMoreFooter.java
package me.simple.loadmoreadapter;

import android.view.View;
import android.widget.ProgressBar;
import android.widget.TextView;

class LoadMoreFooter extends AbsLoadMoreFooter {

    private ProgressBar mProgressBar;
    private TextView mTextView;

    @Override
    public int setLayoutRes() {
        return R.layout.adapter_load_more;
    }

    @Override
    public void onCreate(View footerView) {
        mProgressBar = footerView.findViewById(R.id.load_more_pb);
        mTextView = footerView.findViewById(R.id.load_more_tv);
    }

    @Override
    public void loading() {
        mProgressBar.setVisibility(View.VISIBLE);
        mTextView.setVisibility(View.GONE);
    }

    @Override
    public void loadComplete() {
        mProgressBar.setVisibility(View.GONE);
        mTextView.setVisibility(View.VISIBLE);
        mTextView.setText("加载完成");
    }

    @Override
    public void noMoreData() {
        mProgressBar.setVisibility(View.GONE);
        mTextView.setVisibility(View.VISIBLE);
        mTextView.setText("已无更多数据");
    }

    @Override
    public void loadFailed() {
        mProgressBar.setVisibility(View.GONE);
        mTextView.setVisibility(View.VISIBLE);
        mTextView.setText("加载失败");
    }

}

File: load_more_adapter/src/main/java/me/simple/loadmoreadapter/LoadMoreFooter.kt
package me.simple.loadmoreadapter

import android.view.View
import android.widget.ProgressBar
import android.widget.TextView

/**
 * 默认的底部加载布局
 */
internal class LoadMoreFooter : ILoadMoreFooter {

    override fun setLayoutRes() =  R.layout.adapter_load_more

    override fun onCreate(footerView: View) {
    }

    override fun loading(footerView: View) {
        footerView.findViewById<ProgressBar>(R.id.load_more_pb)?.apply {
            visibility = View.VISIBLE
        }
        footerView.findViewById<TextView>(R.id.load_more_tv)?.apply {
            visibility = View.GONE
        }
    }

    override fun noMoreData(footerView: View) {
        footerView.findViewById<ProgressBar>(R.id.load_more_pb)?.apply {
            visibility = View.GONE
        }
        footerView.findViewById<TextView>(R.id.load_more_tv)?.apply {
            visibility = View.VISIBLE
            text = "已无更多数据"
        }
    }

    override fun loadFailed(footerView: View) {
        footerView.findViewById<ProgressBar>(R.id.load_more_pb)?.apply {
            visibility = View.GONE
        }
        footerView.findViewById<TextView>(R.id.load_more_tv)?.apply {
            visibility = View.VISIBLE
            text = "加载失败"
        }
    }
}